qiskit/circuit/quantumcircuit.py

```diff
@@ -35,6 +35,7 @@
     Mapping,
     Set,
     Iterable,
+    AbstractSet,
```

Comment:
Oh yeah - I originally used `AbstractSet` deliberately because `collections.abc.Set` wouldn't work in Python 3.7, but since this file already has `from __future__ import annotations` and type hints that can't be evaluated in Python 3.7 anyway, I can just change it.

Done in fc0346e61.